debezium-server-pubsub/src/main/java/io/debezium/server/pubsub/PubSubChangeConsumer.java

```diff
@@ -213,12 +214,33 @@ void close() {
                 publisher.shutdown();
             }
             catch (Exception e) {
-                LOGGER.warn("Exception while closing publisher: {}", e);
+                LOGGER.warn("Exception while closing publisher: {}", e.getMessage(), e);
             }
         });
+        shutdownChannel(channel);
+    }
 
+    void shutdownChannel(ManagedChannel channel) {
         if (channel != null && !channel.isShutdown()) {
-            channel.shutdown();
+            ExecutorService executor = Executors.newSingleThreadExecutor();
+            Future<?> future = executor.submit(() -> {
+                channel.shutdown();
+                try {
+                    channel.awaitTermination(channelShutdownTimeout, TimeUnit.SECONDS);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+            });
+
+            try {
+                future.get(channelShutdownTimeout + 1, TimeUnit.SECONDS);
```

Comment:
Timeout is in milliseconds but `get` is in seconds. Also why +1?

There should be also `0` as a special value that will do shutdown as is done currently.